elyra/pipeline/tests/test_kfp_processor.py

```diff
@@ -0,0 +1,165 @@
+#
+# Copyright 2018-2020 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import mock
+import os
+import pytest
+import tarfile
+from minio import Minio
+from elyra.pipeline.processor_kfp import KfpPipelineProcessor
+from elyra.pipeline.pipeline import Operation as Op
+from elyra.pipeline.parser import PipelineParser
+from elyra.util.cos import CosClient
+from elyra.metadata import MetadataManager, Metadata
+from elyra.pipeline.tests.test_pipeline_parser import _read_pipeline_resource
+
+
+@pytest.fixture
+def processor():
+    processor = KfpPipelineProcessor()
+    processor.root_dir = os.getcwd()
+    return processor
+
+
+@pytest.fixture
+def pipeline():
+    pipeline_resource = _read_pipeline_resource('resources/sample_pipelines/pipeline_3_node_sample.json')
+    return PipelineParser.parse(pipeline_resource)
+
+
+def test_fail_get_runtime_configuration(processor):
+    with pytest.raises(RuntimeError):
+        processor._get_runtime_configuration('non_existent_metadata')
+
+
+@pytest.mark.skip(reason='TODO - fix')
+def test_get_runtime_configuration(monkeypatch, processor):
+    print("TODO")
+
+
+def test_generate_dependency_archive(processor):
+    pipelines_test_file = 'elyra/pipeline/tests/resources/archive/test.ipynb'
+    pipeline_dependencies = 'elyra/pipeline/tests/resources/archive/*'
+    correct_filelist = ['test.ipynb', 'airflow.json']
+    test_operation = Op(id='123e4567-e89b-12d3-a456-426614174000',
+                        type='execution-node',
+                        classifier='kfp',
+                        filename=pipelines_test_file,
+                        dependencies=pipeline_dependencies,
+                        runtime_image='tensorflow/tensorflow:latest')
+
+    archive_location = processor._generate_dependency_archive(test_operation)
+
+    tar_content = []
+    with tarfile.open(archive_location, "r:gz") as tar:
+        for tarinfo in tar:
+            if tarinfo.isreg():
+                print(tarinfo.name)
+                tar_content.append(tarinfo.name)
+
+    assert sorted(correct_filelist) == sorted(tar_content)
+
+
+@pytest.mark.skip(reason='test should fail but does not, pipelines_test_file doesnt exist')
+def test_fail_generate_dependency_archive(processor):
+    pipelines_test_file = 'elyra/pipeline/tests/resources/archive/test.txt'
+    pipeline_dependencies = 'elyra/pipeline/tests/resources/archive/*'
```

Comment:
This won't throw any exception because 'include-all' is true - due to the wildcard.  I think we should probably test for the existence of filename prior to the creation of the archive (in the `_generate` method) since it will **always** be required.

Reply:
adding to existing open issue to address #747